joern-cli/frontends/c2cpg/src/test/scala/io/joern/c2cpg/cpp/features20/Cpp20FeaturesTests.scala

```diff
@@ -210,59 +211,103 @@ class Cpp20FeaturesTests extends AstC2CpgSuite(fileSuffix = FileDefaults.CppExt)
       val cpg = code("""
-          |switch (n) {
-          |case 1:
-          |  // ...
-          |  break;
-          |
-          |[[likely]] case 2:  // n == 2 is considered to be arbitrarily more
-          |  // ...            // likely than any other value of n
-          |  break;
-          |}
+          |void foo() {
+          |  switch (n) {
+          |    case 1:
+          |      case1();
+          |      break;
+          |    [[likely]] case 2:  // n == 2 is considered to be arbitrarily more
+          |      case2()           // likely than any other value of n
+          |      break;
+          |  }
           |
-          |int random = get_random_number_between_x_and_y(0, 3);
-          |if (random > 0) [[likely]] {
-          |  // body of if statement
-          |  // ...
-          |}
+          |  int random = get_random_number_between_x_and_y(0, 3);
+          |  if (random > 0) [[likely]] {
+          |    // body of if statement
+          |    likelyIf();
+          |  }
           |
-          |while (unlikely_truthy_condition) [[unlikely]] {
-          |  // body of while statement
-          |  // ...
+          |  while (unlikely_truthy_condition) [[unlikely]] {
+          |    // body of while statement
+          |    unlikelyWhile()
+          |  }
           |}
           |""".stripMargin)
-      ???
+      val cases =
+        cpg.method
+          .nameExact("foo")
+          .controlStructure
+          .controlStructureTypeExact(ControlStructureTypes.SWITCH)
+          .astChildren
+          .isBlock
+          ._jumpTargetViaAstOut
+      cases.code.l shouldBe List("case 1:", "[[likely]] case 2:")
+      cpg.method
+        .nameExact("foo")
+        .controlStructure
+        .controlStructureTypeExact(ControlStructureTypes.SWITCH)
+        .ast
+        .isCall
+        .code
+        .l shouldBe List("case1()", "case2()")
+
+      cpg.method
+        .nameExact("foo")
+        .controlStructure
+        .controlStructureTypeExact(ControlStructureTypes.IF)
+        .ast
+        .isCall
+        .code
+        .l shouldBe List("random > 0", "likelyIf()")
+
+      cpg.method
+        .nameExact("foo")
+        .controlStructure
+        .controlStructureTypeExact(ControlStructureTypes.WHILE)
+        .ast
+        .isCall
+        .code
+        .l shouldBe List("unlikelyWhile()")
     }
 
-    "handle deprecate implicit capture of this" ignore {
+    "handle deprecate implicit capture of this" in {
       val cpg = code("""
           |struct int_value {
           |  int n = 0;
           |  auto getter_fn() {
-          |    // BAD:
-          |    // return [=]() { return n; };
-          |
-          |    // GOOD:
           |    return [=, *this]() { return n; };
           |  }
           |};
           |""".stripMargin)
-      ???
+      // TODO: we can not express these lambda types in the current schema
+      // We would need to add a new type for lambdas that capture `this`
```

Comment:
At the end of the day, we'll want a unique type for every lambda / method reference just like we do it for lambdas in other frontends. I believe differently to those other languages this actually matches C++ semantics, where the return type of `getter_fn` is an "unnamable" lambda type that's unique for each lambda. (But we'd want to do it anyway, because we can't resolve dynamic calls of such functions otherwise.)

But that's just side-commentary, not a blocker.

Reply:
Yes, that's the plan. Just need to figure out how to get the name/fullname for such a lambda during the traversal of the method scope it's located in. At this point during the traversal we have not seen the lambda yet.